src/planscape/plan/models.py

```diff
@@ -70,46 +71,67 @@ class Project(models.Model):
     max_slope: models.FloatField = models.FloatField(null=True)
 
 
+class ConfigPriority(models.Model):
+    # TODO: migrate to Config
+    project = models.ForeignKey(Project, on_delete=models.CASCADE, null=False)
+
+    condition = models.ForeignKey(
+        'conditions.Condition', on_delete=models.CASCADE)
+
+
 class Scenario(models.Model):
     """
-    A Scenario is associated with one User, the owner, and one Project. It has optional user-specified
-    prioritization parameters.
+    A Scenario is associated with one owner, one Plan, and one Project.
+    It contains user specified fields like 'notes'.
     """
     # TODO: Change "null=True" so that owner is not nullable. Currently owner can be null because
     # we want alpha users to not be signed in.
     owner = models.ForeignKey(
         User, on_delete=models.CASCADE, null=True)  # type: ignore
-    
-    plan = models.ForeignKey(Plan, on_delete=models.CASCADE, null=True)  # type: ignore
+
+    plan = models.ForeignKey(
+        Plan, on_delete=models.CASCADE, null=False)  # type: ignore
+
+    project = models.ForeignKey(
+        Project, on_delete=models.CASCADE, null=False)  # type: ignore
 
     # The creation time of the project, automatically set when the project is created.
     creation_time: models.DateTimeField = models.DateTimeField(
         null=True, auto_now_add=True)
 
-    project = models.ForeignKey(
-        Project, on_delete=models.CASCADE, null=True)  # type: ignore
-    
     notes: models.TextField = models.TextField(null=True)
 
-    favorited: models.BooleanField = models.BooleanField(null=True, default=False)
-    
+    favorited: models.BooleanField = models.BooleanField(
+        null=True, default=False)
+
+    class ScenarioStatus(models.IntegerChoices):
+        PENDING = 0
+        PROCESSING = 1
+        PROCESSED = 2
+        FAILED = 3
+
+    status = models.IntegerField(
+        choices=ScenarioStatus.choices, default=ScenarioStatus.PENDING)
+
 
 class ScenarioWeightedPriority(models.Model):
```

Comment:
This is definitely ok if a user creates a config without specifying weights.

How about the situation where a user creates a config and specifies weights in the config (especially if they haven't kicked off the config so no scenario exists yet)? To cover this, can this be generalized to WeightedPriority, and point to both a project/config, and a scenario?

Reply:
My mental model equates setting weights to creating a scenario, no matter when it happens in the flow. So I think behind the scenes, we could create a scenario and then when the user clicks "generate scenario", use that same one to retrieve the weights. I don't have a strong preference, other than having a lot of FKs might make it unclear what the weights are actually assoc w.

Reply:
I see!
My model is a little different - a scenario is only created when a user kicks off a config.

If we move forward with your model, could you also add a scenario status that's "wip", "not yet kicked off", or something along those lines? It seems different from "pending" and I want to make sure such scenarios won't show up when the FE requests a list of scenarios.

Reply:
that makes sense! added one for INITIALIZED